ts/WoltLabSuite/Core/Acp/Ui/Package/Update/Manager.ts

```diff
@@ -0,0 +1,200 @@
+/**
+ * Allows the selection of packages for an update and handles any possible
+ * conflicts. Prompts for credentials if necessary.
+ *
+ * @author      Alexander Ebert
+ * @copyright   2001-2022 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module      WoltLabSuite/Core/Acp/Ui/Package/Update/Manager
+ */
+
+import { dboAction } from "../../../../Ajax";
+import { DialogCallbackObject, DialogCallbackSetup } from "../../../../Ui/Dialog/Data";
+import * as Language from "../../../../Language";
+import * as UiDialog from "../../../../Ui/Dialog";
+import DomUtil from "../../../../Dom/Util";
+
+const enum ResponseType {
+  AuthorizationRequired = "authorizationRequired",
+  Conflict = "conflict",
+  Queue = "queue",
+}
+
+type ResponsePrepareUpdate =
+  | {
+      template: string;
+      type: ResponseType.AuthorizationRequired;
+    }
+  | {
+      template: string;
+      type: ResponseType.Conflict;
+    }
+  | {
+      queueID: number;
+      type: ResponseType.Queue;
+    };
+
+type AuthData = {
+  packageUpdateServerID: number;
+  password: string;
+  saveCredentials: boolean;
+  username: string;
+};
+
+type Payload = {
+  authData?: AuthData;
+  packages: Record<string, string>;
+};
+
+class AcpUiPackageUpdateManager implements DialogCallbackObject {
+  private readonly packages = new Map<HTMLElement, HTMLInputElement | null>();
+  private readonly submitButton: HTMLButtonElement;
+
+  constructor() {
+    this.submitButton = document.getElementById("packageUpdateSubmitButton") as HTMLButtonElement;
+
+    this.submitButton.addEventListener("click", (event) => {
+      event.preventDefault();
+
+      void this.prepareUpdate();
+    });
+
+    this.initPackages();
+  }
+
+  private initPackages(): void {
+    document.querySelectorAll(".packageUpdate").forEach((element: HTMLElement) => {
+      const checkbox = element.querySelector<HTMLInputElement>('input[type="checkbox"]');
+      checkbox?.addEventListener("change", () => this.updateSubmitButtonState());
+
+      this.packages.set(element, checkbox);
+    });
+  }
+
+  private updateSubmitButtonState(): void {
+    let disableButton = true;
+    this.packages.forEach((checkbox) => {
+      if (!checkbox || checkbox.checked) {
+        disableButton = false;
+      }
+    });
```

Comment:
This technically could be `.every(checkbox => checkbox?.checked)`, however that would require an additional `Array.from()`. I leave this up to you.

Reply:
This isn't equal, because `checkbox` might be `null` and thus it is implicitly checked. It would be much cleaner if I could do a return inside the condition. I don't think that using an array makes it much cleaner.

Reply:
> This isn't equal, because checkbox might be null and thus it is implicitly checked.

Ah, indeed. I've messed up the condition when doing De Morgan's rule in my head. The suggestion primarily was about the `.every()` instead of `.forEach()`, but it's not a clear-cut case here. So leaving this up to you.

Reply:
I'm leaving it as-is, because I wouldn't favor one over the other here.